packages/build.gradle.kts

```diff
@@ -33,6 +33,136 @@ allprojects {
     }
 }
 
+/**
+ * Task that will build and publish the defined packages to <root>/packages/build/m2-buildrepo`.
+ * This is mostly suited for CI jobs that wants to build select publications on specific runners.
+ *
+ *
+ * See `gradle.properties` for specific configuration options available to this task.
+ *
+ * For local development, using:
+ *
+ * ```
+ * > ./gradlew publishAllPublicationsToTestRepository
+ * ```
+ *
+ * will build and publish all targets available to the builder platform.
+ */
+tasks.register("publishCIPackages") {
+    group = "Publishing"
+    description = "Publish packages that has been configured for this CI node. See `gradle.properties`."
+
+    // Figure out which targets are configured. This will impact which sub modules will be published
+    val availableTargets = setOf(
+        "iosArm64",
+        "iosX64",
+        "jvm",
+        "macosX64",
+        "macosArm64",
+        "android",
+        "metadata",
+        "compilerPlugin",
+        "gradlePlugin"
+    )
+
+    val mainHostTarget: Set<String> = setOf("metadata") // "kotlinMultiplatform"
+
+    val isMainHost: Boolean? = if (project.properties.containsKey("realm.kotlin.mainHost"))  {
+        project.properties["realm.kotlin.mainHost"] == "true"
+    } else {
+        null
+    }
+    // Find user configured platforms (if any)
+    val userTargets: Set<String>? = (project.properties["realm.kotlin.targets"] as String?)
+        ?.split(",")
+        ?.map { it.trim() }
+        ?.filter { it.isNotEmpty() }
+        ?.toSet()
+    userTargets?.forEach {
+        if (!availableTargets.contains(it)) {
+            project.logger.error("Unknown publication: $it")
+            throw IllegalArgumentException("Unknown publication: $it")
+        }
+    }
+    // Configure which platforms publications we do want to publish
+    val wantedTargets: Collection<String> = when (isMainHost) {
```

Comment:
This logic is not that clear. Wouldn't you also have to filter out `mainHostTarget` from `userTargets` if `isMainHost==false` to be safe.  

Maybe just define a default for `isMainHost` and then have
```
val targets = (userTargets?: availableTargets).let {
   when (isMainHost) {
     true -> it + mainHostTarget
     false -> it - mainHostTarget
  }
}  
``` 
Otherwise maybe add some comments, so that it is clear what we expect.